bril-txt/briltxt.py

```diff
@@ -114,13 +136,82 @@ def bool(self, items):
     def type(self, items):
         return str(items[0])
 
-
-def parse_bril(txt):
+def parse_helper(txt):
     parser = lark.Lark(GRAMMAR)
     tree = parser.parse(txt)
     data = JSONTransformer().transform(tree)
+    return data
+
+# Import dependencies of imported functions as well
+
+def func_walk(imports, statements):
+    dependencies = imports
+    imported = []
+    data = []
+    while(dependencies):
+        dependency = dependencies.pop(0)
+        statement_pointer = 0
+
+        while (statement_pointer != len(statements)):
+            statement = statements[statement_pointer]
+            if 'instrs' in statement and \
+             dependency['name'] == statement['name'] and \
+             dependency['alias'] not in imported:
+                statement['name'] = dependency['alias']
+                imported.append(statement['name'])
+                data.append(statement)
+                for instr in statement['instrs']:
+                    if instr['op'] == 'call':
+                        dependencies.append({'name': instr['args'][0], 'alias': instr['args'][0]})
+                statement_pointer = len(statements)
+            else:
+                statement_pointer += 1
+    return data
+
+def unroll_imports(txt, modules):
+    stage_one = parse_helper(txt)
+
+    imports = {}
+    for statement in stage_one['functions']:
+        if "import" in statement.keys():
+            imports[statement['import']] = statement['functionids']
+
+    for imp in imports.keys():
```

Comment:
Just use `imports`, no need for `imports.keys(). You can also do `for imp, val in imports` to get the value as well, without having to do `imports[imp]`.